cfme/infrastructure/virtual_machines.py

```diff
@@ -667,6 +680,13 @@ def get_changes_to_fill(self, other_configuration):
                 changes['disks'].append({'action': 'delete', 'disk': disk, 'delete_backing': None})
             elif disk not in self.disks and disk in other_configuration.disks:
                 changes['disks'].append({'action': 'add', 'disk': disk})
+            elif disk in self.disks and disk in other_configuration.disks:
+                old_disk = [d for d in self.disks if d == disk][0]
+                new_disk = [d for d in other_configuration.disks if d == disk][0]
```

Comment:
[Optional] 
it would look better to me to use here something like
next(d for d in self.disks if d == disk)
it won't generate full list, just find first value :)